src/main/java/org/spongepowered/api/event/world/ChangeWorldBorderEvent.java

```diff
@@ -48,6 +48,16 @@
      */
     Optional<WorldBorder> previousBorder();
 
+    /**
+     * Gets the border that was originally meant to be used as an {@link Optional}.
+     * For some targets, the border may be unset and therefore they may not always
+     * have a border. It is in these instances that {@link Optional#empty()} is
+     * returned.
+     *
+     * @return The border that will be in use, if there is one.
+     */
+    Optional<WorldBorder> originalNewBorder();
```

Comment:
At this point, should we use Transaction? Since it manages the state of original, custom, default, and can always view the final replacement (along with invalidate, isValid, etc. )

Reply:
Given that an entry here can be null and Transactions does not allow for null values - no.